### PR TITLE
Fix Images Version Sorting

### PR DESCRIPTION
**Pull request by tonysm**
### Changed

- Fixes the version sorting so we consider semver image tags higher than the other distro-specific or flavored image tag names

---

closes https://github.com/tighten/takeout/issues/337

<details>
<summary>Screenshots</summary>

typesense
sqs
solr
singlestore
redis
postgres
postgis
opensearch
neo4j
mysql
mssql
mongo
minio
memcached
meilisearch
mariadb
mailpit
maildev
influxdb
expose
eventstoredb
elasticsearch

</details>

### CODE REVIEW

File: tests/Feature/VersionComparatorTest.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+use App\Shell\VersionComparator;
+use Tests\TestCase;
+
+class VersionComparatorTest extends TestCase
+{
+    /**
+     * @test
+     *
+     * @testWith [["latest", "8.0"], ["latest", "8.0"]]
```
Comment (gcavanunez) [inline tests/Feature/VersionComparatorTest.php:13]: Nice!

**Review by gcavanunez — APPROVED** [review]
This is awesome! ran it locally, lists were sorting nicely, I feel like this really opens things up to turning this field into a combox/search select terminal prompt

**Review by jakebathman — APPROVED** [review]
This all looks good, nice work getting it more simplified and using that new comparator class. I haven't run this locally but if Guillermo has then that's great.

**Comment by tonysm** [comment]
I've also tested this on an M1 and it works. One thing I want to eventually handle is that it should consider `8.0` as higher than `8.0.45`, but that can be enhanced later.

**Pull request merged**